VERSIONING.md

```diff
@@ -0,0 +1,145 @@
+segment](https://semver.org/#spec-item-9). Pre-release information is used
+instead of [build metadata](https://semver.org/#spec-item-10) as pre-release
+precedence in honored in semantic versioning. The `<components>` numeric
+identifier only corresponds to the version of components for its preceding
+`<major>`, `<minor>`, `<patch>` versions. There is no relationship between
+`<components>` versions across `<major>`, `<minor>`, `<patch>` versions.
+
+An example version string for a UXP release that embeds Crossplane `v1.3.0`
+could be:
+
+```
+v1.3.0-up.2
+```
+
+Additional pre-release segments may be appended both before and after the
+`-up-<components>` segment. Any pre-release segments appended before will
+correspond to the embedded Crossplane version (e.g. `v1.4.0-rc.0-up.1`). Any
+pre-release segments appended after will correspond to a pre-release version of
+UXP itself (e.g. `v1.4.0-up.1-rc.0`). There should _never_ be both a Crossplane
+pre-release and UXP pre-release on the same version string (see [Guarantees
+section](#guarantees)).
+
+## Terms
+
+The following terms are used throughout this document and are defined as
+follows:
+
+- **UXP Version**: the entire semantic version for a UXP release (e.g.
+  `v1.2.2-up.3`).
+- **Crossplane Version**: the version of Crossplane embedded in the UXP release
+  (e.g. `v1.2.2`).
+- **Components Marker**: the version of the components surrounding upstream (or
+  temporarily forked) Crossplane (e.g. `-up.3`).
+
+## Guarantees
+
+![versioning-diagram](docs/media/versioning.png)
+
+The UXP versioning strategy can be expressed through a series of guarantees,
+each of which are elaborated in the subsequent sections. The diagram above
+illustrates examples of each guarantee in practice.
+
+1. The components marker on UXP main should always end in `-up.1` (i.e. no
+   components release candidates).
+2. The Crossplane version on UXP main must always be a release candidate version
+   for the first release of a major/minor pair (i.e. `vX.Y.0-rc.Z`).
+3. The Crossplane version on any `release-*` branch should _always_ be a
+   Crossplane release version (e.g. `v1.2.2`) (i.e. no release candidates).
+4. Not every Crossplane patch for a given `<major>`/`<minor>` version must be
+   included in a UXP release, even if the `<major>`/`<minor>` version has been
+   selected for long-term service (LTS). This would typically occur if
+   Crossplane made multiple patch releases in quick succession and a UXP version
+   had not been released for each in the intermediate (e.g. if `v1.2.4` is
+   released in upstream Crossplane and has not been included in a UXP release
+   before `v1.2.5` is released upstream, UXP may elect to forego cutting a
+   release with `v1.2.4`).
+5. At least one release candidate (RC) version should exist for any version that
+   is released. This is typically accomplished by cutting the RC version for the
+   _next_ release immediately after release of the preceding version.
```

Comment:
Would this mean we will be tagging the same commit first with the released version and then after the release with the RC version? Does our build tooling handle multiple tags on the same commit properly (e.g. use the latest applied tag) or would we need a dummy commit to move the head of the release branch before applying RC tag?

Reply:
Interestingly our build tooling does handle tags on the same commit (it will use the most recent by timestamp), but I am an advocate for merging an empty commit and tagging it instead. This is how we advance branches upstream (ref https://github.com/crossplane/crossplane/pull/2395).

Reply:
Do you think if it makes sense to document this here?

Reply:
Done 👍🏻